lib/core-classes/DecodersService.ts

```diff
@@ -1,31 +1,70 @@
-import { PluginContext, EmbeddedSDK, Plugin } from "kuzzle";
+import { DecoderQueue } from './../types/decoders/decodersQueue';
+import { PluginContext, EmbeddedSDK, Plugin, Inflector, PluginImplementationError, PluginApiDefinition } from "kuzzle";
 
 import { Decoder } from "./Decoder";
+import { PayloadHandler, PayloadService } from './PayloadService';
 import { DeviceManagerConfig } from "../DeviceManagerPlugin";
-import { DecoderContent } from '../types/decoders/DecodersContent';
+import { DecoderContent } from "../types/decoders/DecodersContent";
 
 export class DecodersService {
   private config: DeviceManagerConfig;
   private context: PluginContext;
+  private api: PluginApiDefinition;
+  private payloadService: PayloadService;
 
   private decoders: Map<string, Decoder>;
 
   get sdk(): EmbeddedSDK {
     return this.context.accessors.sdk;
   }
 
-  constructor(plugin: Plugin, decoders: Map<string, Decoder>) {
+  constructor(plugin: Plugin, decoders: Map<string, Decoder>, decodersQueue: DecoderQueue[], payloadService: PayloadService) {
     this.config = plugin.config as any;
     this.context = plugin.context;
+    this.api = plugin.api;
+    this.payloadService = payloadService;
 
     this.decoders = decoders;
+
+    decodersQueue.forEach(({ decoder, handler }) => {
+      this.register(decoder, { handler });
+      this.context.log.info(`Register API action "device-manager/payload:${decoder.action}" with decoder "${decoder.constructor.name}" for device "${decoder.deviceModel}"`);
+    });
   }
 
   async list(): Promise<DecoderContent[]> {
-    const decoders = Array
-      .from(this.decoders.values())
-      .map(decoder => decoder.serialize())
+    const decoders = Array.from(this.decoders.values()).map((decoder) =>
+      decoder.serialize()
+    );
 
     return decoders;
   }
+
+  register(
+    decoder: Decoder,
+    { handler }: { handler?: PayloadHandler } = {}
+  ): { controller: string; action: string } {
+    decoder.action = decoder.action || Inflector.kebabCase(decoder.deviceModel);
+
+    if (this.api["device-manager/payload"].actions[decoder.action]) {
+      throw new PluginImplementationError(
+        `A decoder for "${decoder.deviceModel}" has already been registered.`
+      );
+    }
+
+    this.api["device-manager/payload"].actions[decoder.action] = {
```

Comment:
This break the single responsibility principle, this class should be aware of other classes internal mechanisms (here it's the plugin API definition from the main Plugin class).

We can discuss about possible solutions if you want